npm/qsharp/src/shared/circuit.ts

```diff
@@ -3,15 +3,34 @@
 
 import { Register } from "./register.js";
 
+/**
+ * Current format version.
+ */
+export const CURRENT_VERSION = "1.0.0";
+
 /**
  * Circuit to be visualized.
  */
 export interface Circuit {
+  version: string;
   /** Array of qubit resources. */
   qubits: Qubit[];
-  operations: Operation[];
+  operations: Operation[][];
```

Comment:
In general, I dislike array-of-array as a type... for one, multi-dimensional arrays are just annoying to code against. But also, there's usually an intermediate type in there you're hiding, that deserves its own type for clarity.

What does an `Operation[]` represent? A column, I assume? You can wrap that it its own object:

```ts
interface Column { 
    operations: Operation[];
}
```

This also makes the type more future-proof in case you ever want to add a new property to the column itself (like, what if columns could have annotations? I don't know, just making stuff up)

And then this property here can become
```ts
    columns: Column[];
```

Reply:
done